core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -219,6 +220,19 @@ private[history] class FsHistoryProvider(conf: SparkConf, clock: Clock)
     }
   }
 
+  override def getEventLogPaths(appId: String, attemptId: Option[String]): Seq[Path] = {
+
+    val filePaths = new ArrayBuffer[Path]()
+    applications.get(appId).foreach { appInfo =>
```

Comment:
The reason I left `attempt.attemptId.isEmpty` here is so that even if someone makes a bad request asking for logs for an attempt which has no `attemptId`, we still download all logs. I know it is a bad request, but it is easy to handle.  Also, I am not sure how `==` works for an `Option`, so this ensure that an exception does not get thrown in the `get` calls at the end of the line.